test/FunctionalTests/Microsoft.OData.Core.Tests/Json/JsonWriterTests.cs

```diff
@@ -175,6 +215,116 @@ private void VerifyWriterPrimitiveValueWithIeee754Compatible<T>(T parameter, str
 
         #endregion
 
+        #region JsonWriter Extension Methods
+
+        [Fact]
+        public void WriteJsonObjectValueWritesJsonObject()
+        {
+            var properties = new Dictionary<string, object>
+            {
+                { "Name", "Sue" },
+                { "Attributes",
+                    new Dictionary<string, object>
+                    {
+                        { "Height", 1.77 },
+                        { "Weight", 80.7 }
+                    }
+                }
+            };
+            this.writer.WriteJsonObjectValue(properties, null);
+            Assert.Equal("{\"Name\":\"Sue\",\"Attributes\":{\"Height\":1.77,\"Weight\":80.7}}", this.builder.ToString());
+        }
+
+        [Fact]
+        public void WriteJsonObjectValueCallsInjectedPropertyAction()
+        {
+            var properties = new Dictionary<string, object> { { "Name", "Sue" } };
+            Action<IJsonWriter> injectPropertyDelegate = (IJsonWriter actionWriter) =>
+            {
+                actionWriter.WriteName("Id");
+                actionWriter.WriteValue(7);
+            };
+
+            this.writer.WriteJsonObjectValue(properties, injectPropertyDelegate);
+            Assert.Equal("{\"Id\":7,\"Name\":\"Sue\"}", this.builder.ToString());
+        }
+
+        [Fact]
+        public void WriteJsonObjectValueWritesPrimitiveCollection()
+        {
+            var properties = new Dictionary<string, object>
+            {
+                { "Names", new string[] { "Sue", "Joe", null } }
+            };
+
+            this.writer.WriteJsonObjectValue(properties, null);
+            Assert.Equal("{\"Names\":[\"Sue\",\"Joe\",null]}", this.builder.ToString());
+        }
+
+        [Fact]
+        public void WriteODataValueWritesODataValue()
+        {
+            var value = new ODataPrimitiveValue(3.14);
+            this.writer.WriteODataValue(value);
+            Assert.Equal("3.14", this.builder.ToString());

```

Comment:
```suggestion
            var input = 3.14;
            var value = new ODataPrimitiveValue(input);
            this.writer.WriteODataValue(value);
            Assert.Equal(input.ToString(), this.builder.ToString());
```

There might be something I'm missing about floats or what the test is trying to accomplish